### PR TITLE
Vendor gardener/gardener@v1.50.1

### PR DESCRIPTION
**Pull request by dkistner**
**How to categorize this PR?**
/kind enhancement
/platform openstack

**What this PR does / why we need it**:
Vendor `gardener/gardener@v1.50.1` to be in sync with the other provider-extensions.

**Special notes for your reviewer**:
Infrastructure integration test succeed locally.

**Release note**:

```other operator
The following dependencies are updated:
- github.com/gardener/gardener: v1.48.0 -> v1.50.1
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -131,34 +132,35 @@ require (
...
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.24.2
+	k8s.io/apimachinery => k8s.io/apimachinery v0.24.2
+	k8s.io/apiserver => k8s.io/apiserver v0.24.2
+	k8s.io/autoscaler => k8s.io/autoscaler v0.0.0-20220531185024-cc90d57b7fe1 // translates to k8s.io/autoscaler/vertical-pod-autoscaler@v0.11.0
+	k8s.io/autoscaler/vertical-pod-autoscaler => k8s.io/autoscaler/vertical-pod-autoscaler v0.11.0
+	k8s.io/client-go => k8s.io/client-go v0.24.2
+	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.24.2
+	k8s.io/code-generator => k8s.io/code-generator v0.24.2
+	k8s.io/component-base => k8s.io/component-base v0.24.2
 	k8s.io/helm => k8s.io/helm v2.13.1+incompatible
```
Comment (MartinWeindel) [inline go.mod:161]: You may drop this line, as in line 137 helm v2.16.1 is already required (by gardener)

**Review by MartinWeindel — DISMISSED** [review]
/lgtm

Only a single comment about helm version (not important).

**Comment by acumino** [comment]
Please adapt HVPA similar to https://github.com/gardener/gardener-extension-provider-azure/pull/541

**Comment by dkistner** [comment]
@acumino, @MartinWeindel addressed the feedback. ptal :)

**Review by acumino — APPROVED** [review]
/lgtm

**Pull request merged**